hack/e2e.go

```diff
@@ -207,6 +207,11 @@ func run(deploy deployer) error {
 	}
 
 	if *up {
+		// If we tried to bring the Kubemark cluster up, make a courtesy
+		// attempt to bring it down so we're not leaving resources around.
+		if *down {
+			defer xmlWrap("TearDown", deploy.Down)
+		}
 		// Start the cluster using this version.
 		if err := xmlWrap("Up", deploy.Up); err != nil {
 			return fmt.Errorf("starting e2e cluster: %s", err)
```

Comment:
Yes - I completely don't understand your suggestion Erik. I'm duplicating the logic that we had in the past:
https://github.com/fejta/kubernetes/blob/ffe611569dbd645a7d5c9428828ece1404e7e396/hack/jenkins/e2e-runner.sh#L169-L181
and that logic worked correctly. However, your PR https://github.com/kubernetes/kubernetes/pull/28575 broke it. I think it's better to have something working (potentially not ideal) than not-working. I don't have time to tune the whole framework.